primitives/src/types.rs

```diff
@@ -116,3 +117,28 @@ impl<AssetId: Copy, Balance: Copy> FeeExchange<AssetId, Balance> {
 		}
 	}
 }
+
+// the following types are shared by crml staking and rewards.
+
+/// The amount of exposure (to slashing) than an individual nominator has.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug)]
+pub struct IndividualExposure<AccountId, Balance: HasCompact> {
+	/// The stash account of the nominator in question.
+	pub who: AccountId,
+	/// Amount of funds exposed.
+	#[codec(compact)]
+	pub value: Balance,
+}
+
+/// A snapshot of the stake backing a single validator in the system.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default, RuntimeDebug)]
+pub struct Exposure<AccountId, Balance: HasCompact> {
```

Comment:
The same here, the types that could be better located in the Staking pallet now moved to be next to other unrelated stuff.

Reply:
The calculations could be, the main reason for separating it like this to isolate the reward related storage and config from the staking storage and config, it made sense to bring the calculations also.
+ the staking module is on the big side already

Reply:
We could even separate the storage and config in the same way that we have done for Sylo within the same pallet. At the end of the day when a service like staking is complex, it's true that we should decompose it to smaller components but we don't necessarily need to then scatter those components from the only location they can have a meaningful place. If there was a different pallet or service that could run with rewards but without the staking module, it could make sense. But now I'm not sure.

Reply:
Is this suggestion about where the rust files/modules sit? e.g.
```
# now
crml/rewards
crml/staking
# after
crml/staking/
crml/staking/rewards/<reward crate>
```
if so then yes I should be able to make this change

Reply:
Yup correct